### PR TITLE
Problem getting test for game create to pass

### PR DESCRIPTION
**Pull request by dennisgit42**
I am still working on getting a basic Game setup, and wrote a RSpec test to see if my GamesController will create a game and then redirect the user to the games#show page. (I also modified the routes to add :show action). However, I keep getting one failing test, and I at a loss as to how to fix it since I am sure I have the correct routes, and provided the necessary info in the games controller for it find game I am creating in my games_controller_spec.rb file. Can you take a look?

### CODE REVIEW

**Comment by dennisgit42** [comment]
Here is the error I get when I run bundle exec rspec:
`[Web Dev]:  /vagrant/src/chess $ bundle exec rspec
F.

Failures:

  1) GamesController games#create action should successfully create new game in the database
     Failure/Error: redirect_to game_path(@game)

     ActionController::UrlGenerationError:
       No route matches {:action=>"show", :controller=>"games", :id=>nil} missing required keys: [:id]
     # ./app/controllers/games_controller.rb:6:in `create'
     # ./spec/controllers/games_controller_spec.rb:6:in `block (3 levels) in <top (required)>'

Finished in 0.06475 seconds (files took 3 seconds to load)
2 examples, 1 failure

Failed examples:

rspec ./spec/controllers/games_controller_spec.rb:5 # GamesController games#create action should successfully create new game in the database
`

File: spec/controllers/games_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe GamesController, type: :controller do
+  describe "games#create action" do
+    it "should successfully create new game in the database" do
+      post :create, params: { game: {game_id: 1, user_white_id: 1, user_black_id: 2} }
```
Comment (mscoutermarsh) [inline spec/controllers/games_controller_spec.rb]: When creating a new game, Rails will autoassign the `id`, so you don't need to pass `game_id: 1` here.

File: spec/controllers/games_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+      expect(response).to redirect_to game_path
```
Comment (mscoutermarsh) [inline spec/controllers/games_controller_spec.rb]: `game_path` expects you to pass an object to it (which is why you're getting the error).

Can do this.

 `expect(response).to redirect_to game_path(Game.last)`

Comment (mscoutermarsh) [inline spec/controllers/games_controller_spec.rb]: I'd also add in a couple tests to verify the correct attributes are being set.

Such as

`expect(Game.last.user_white_id).to eq 1`

**Comment by mscoutermarsh** [comment]
Added some ideas for fixing the test. Let me know how it turns out 😄

**Comment by dennisgit42** [comment]
I am still getting the same error message after making those changes. Also, I am getting error when I try to create a new game in rails console:
`
irb(main):001:0> game = Game.create({user_white_id: 1, user_black_id:2})
   (0.1ms)  BEGIN
   (0.2ms)  ROLLBACK
=> #<Game id: nil, created_at: nil, updated_at: nil, user_white_id: 1, user_black_id: 2>
`

**Comment by mscoutermarsh** [comment]
@dennisgit42 After running that command.

Run `game.errors` and it will show you why it failed.

**Comment by dennisgit42** [comment]
So I am still getting the same error I started out with:
`
[Web Dev]:  /vagrant/src/chess $ bundle exec rspec
F.

Failures:

  1) GamesController games#create action should successfully create new game in the database
     Failure/Error: expect(response).to redirect_to game_path(Game.last)

     ActionController::UrlGenerationError:
       No route matches {:action=>"show", :controller=>"games", :id=>nil} missing required keys: [:id]
     # ./spec/controllers/games_controller_spec.rb:9:in `block (3 levels) in <top (required)>'

Finished in 0.15291 seconds (files took 3.17 seconds to load)
2 examples, 1 failure

Failed examples:

rspec ./spec/controllers/games_controller_spec.rb:5 # GamesController games#create action should successfully create new game in the database
`
I just pushed my latest changes to my branch named 'dennis'

**Comment by dennisgit42** [comment]
Finished creation of new board with help of mentor, all rspec tests pass

**Comment by dennisgit42** [comment]
Also, I don't know why Travis is giving us a bunch of failing tests saying that 'Environment Data not found in schema"  To resolve this issue, run: 
        bin/rails db:environment:set RAILS_ENV=test
I tried running that command in my web dev window and it apparently does nothing, and leaves nothing new for me to commit to my branch.

**Comment by dennisgit42** [comment]
All travis tests now pass, now we just need code review and resolving the merge conflicts.

File: app/controllers/games_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class GamesController < ApplicationController
+  before_action :authenticate_user!, only: %i[new create show]
+
+  def new; end
+
+  def create
+    @game = Game.new(game_params)
+    @game.save
```
Comment (mscoutermarsh) [inline app/controllers/games_controller.rb]: `@game = Game.create(game_params)`

Is the same as using `new` and then `save`

File: spec/models/game_spec.rb [hunk 4]
```diff
@@ -1,5 +1,65 @@
...
+    end
+
+    it 'creates one king for each side' do
+      game = FactoryGirl.create(:game)
+      game.populate_board!
+      n_white_kings = game.pieces.where(type: KING, is_white: true).count
+      expect(n_white_kings).to eq(1)
+      n_black_kings = game.pieces.where(type: KING, is_white: false).count
+      expect(n_black_kings).to eq(1)
+    end
```
Comment (mscoutermarsh) [inline spec/models/game_spec.rb]: nice!

**Pull request merged**